tests/e2e/login/privacy-policy.wdio-spec.js

```diff
@@ -0,0 +1,128 @@
+  const frenchTexts = privacyPolicyFactory.french;
+  const offlineUser = userFactory.build();
+  const onlineUser = userFactory.build({
+    username: 'online',
+    roles: ['program_officer'],
+    place: {
+      _id: 'hc2',
+      type: 'health_center',
+      name: 'Health Center 2',
+      parent: 'dist1'
+    },
+    contact: {
+      _id: 'fixture:user:onlineuser',
+      name: 'onlineuser'
+    }
+  });
+
+  const parent = placeFactory.place().build({_id:'dist1',type:'district_hospital'});
+
+  before(async () => {
+    await utils.saveDocs([parent, privacyPolicyFactory.privacyPolicy().build()]);
+    await utils.createUsers([ onlineUser, offlineUser]);
+  });
+
+  describe('for an online user', () => {
+    it('should show the correct privacy policy on login', async () => {
+      // After first login, check that privacy policy was prompted to user
+      await loginPage.login(onlineUser.username, onlineUser.password);
+      await browser.waitUntil(async () => {
+        const wrapperText = await (await privacyPolicyPage.privacyWrapper()).getText();
+        return wrapperText.includes(englishTexts.header) && wrapperText.includes(englishTexts.paragraph);
+      }, 'Timed out waiting for English Online Privacy Policy to Display');
+
+      // After accepting, no privacy policy on next load
+      await privacyPolicyPage.acceptPrivacyPolicy();
+      await browser.url('/');
+      expect(await privacyPolicyPage.privacyWrapper()).not.toBeDisplayed();
+
+      // Check display when loading privacy policy page
+      await privacyPolicyPage.goToPrivacyPolicyConfig();
+      await browser.waitUntil(async () => {
+        const wrapperText = await (await privacyPolicyPage.privacyConfig()).getText();
+        return wrapperText.includes(englishTexts.header) && wrapperText.includes(englishTexts.paragraph);
+      }, 'Timed out waiting for english online privacy to display');
+
+      // No privacy policy on 2nd login
+      await browser.reloadSession();
+      await browser.url('/');
+      await loginPage.login(onlineUser.username, onlineUser.password);
+      expect(await privacyPolicyPage.privacyWrapper()).not.toBeDisplayed();
+
+
+      // After login in french, check that privacy policy was prompted to user again
+      await browser.reloadSession();
+      await browser.url('/');
+      await loginPage.login(onlineUser.username, onlineUser.password, 'fr');
+      await browser.waitUntil(async () => {
+        const wrapperText = await (await privacyPolicyPage.privacyWrapper()).getText();
+        return wrapperText.includes(frenchTexts.header) && wrapperText.includes(frenchTexts.paragraph);
+      }, 'Timed out waiting for french online privacy to display');
+      await privacyPolicyPage.acceptPrivacyPolicy();
+    });
+  });
+
+
+  // WDIO currently cannot log in offline users because of service worker race condition
+  // https://github.com/medic/cht-core/issues/7242
+  xdescribe('for a french offline user', () => {
```

Comment:
Can we please enable the tests for offline users on CI? I haven't seen the race condition in CI.

Reply:
I feel real uneasy about it as I can't get the other restricted users tests to pass locally or this one. The feed back cycle wouldn't be great. As in I make a change and then push it to run and if it fails I repeat the cycle but I could disable everything else and just run this test it's still unmaintainable in my opinion.